app.rb

```diff
@@ -0,0 +1,57 @@
+require './person'
+require './book'
+require './rental'
+require './teacher'
+require './student'
+
+# app file to hold all operations
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "#{book.id}: #{book.title} by #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "#{person.id}: #{person.name}"
+    end
+  end
+
+  def create_person(type, id, name, parent_permission)
+    if type == 'teacher'
+      @people << Teacher.new(id, name)
+    elsif type == 'student'
+      @people << Student.new(id, name, parent_permission: parent_permission)
+    else
+      puts 'Invalid person type.'
+    end
```

Comment:
After trying to create a person, first I did not get a success message to indicate that I was successful. Secondly, then I tried to list all people but  I got unknown. 

<img width="351" alt="image" src="https://github.com/Daniel-Kigozi/school_library/assets/84156714/be3a60b3-46b9-43d0-8ea0-a8f54f30535d">

In the image above, at point `A` I choose the option to create a teacher, then at point `B` I input the name, after this point there should be a success message. Then at point `C`, I choose the option to list all people. I should have the name of the teacher I just created but, I got an unknown at point `D`.